src/aws-cpp-sdk-core/include/smithy/interceptor/impl/ChecksumInterceptor.h

```diff
@@ -0,0 +1,209 @@
+/**
+* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#pragma once
+#include <smithy/interceptor/Interceptor.h>
+
+#include <aws/core/utils/crypto/PrecalculatedHash.h>
+#include <aws/core/utils/crypto/CRC32.h>
+#include <aws/core/utils/crypto/Sha1.h>
+#include <aws/core/utils/crypto/Sha256.h>
+#include <aws/core/utils/HashingUtils.h>
+
+namespace smithy
+{
+    namespace interceptor
+    {
+        static const char* CHECKSUM_INTERCEPTOR_LOG_TAG = "ChecksumInterceptor";
+
+        class ChecksumInterceptor final : public Interceptor
+        {
+        public:
+
+            ChecksumInterceptor() = default;
+            ChecksumInterceptor(const ChecksumInterceptor& other) = delete;
+            ChecksumInterceptor(ChecksumInterceptor&& other) noexcept = default;
+            ChecksumInterceptor& operator=(const ChecksumInterceptor& other) = delete;
+            ChecksumInterceptor& operator=(ChecksumInterceptor&& other) noexcept = default;
+            ~ChecksumInterceptor() override = default;
+
+            ModifyRequestOutcome ModifyBeforeSigning(InterceptorContext& context) override
+            {
+                const auto& httpRequest = context.GetTransmitRequest();
+                const auto& modeledRequest = context.GetModeledRequest();
+                if (httpRequest == nullptr)
```

Comment:
nit : can also use  if (!httpRequest)

Reply:
[this is the code style that crt uses](https://github.com/search?q=repo%3Aawslabs%2Faws-crt-cpp%20%22%3D%3D%20nullptr%22&type=code) so i wanted to keep it consistent

Reply:
if thats the case, its my advice adopt  this to avoid typo '='
if (nullptr == httpRequest)